src/pages/audio-rewards-page/components/modals/ChallengeRewards.tsx

```diff
@@ -162,19 +162,10 @@ const ChallengeRewardsBody = ({ dismissModal }: BodyProps) => {
     modalButtonInfo
   } = challengeRewardsConfig[modalType]
 
-  const currentStepCountOverrides = useOptimisticChallengeCompletionStepCounts()
-  const currentStepCountOverride = currentStepCountOverrides[modalType]
-  const shouldOverrideCurrentStepCount =
-    !challenge?.is_complete && currentStepCountOverride !== undefined
-  const currentStepCount = shouldOverrideCurrentStepCount
-    ? currentStepCountOverride!
-    : challenge?.current_step_count || 0
-
+  const currentStepCount = challenge?.current_step_count || 0
   const isIncomplete = currentStepCount === 0
   const isInProgress = currentStepCount > 0 && currentStepCount < stepCount
-  const isComplete = shouldOverrideCurrentStepCount
-    ? currentStepCountOverride! >= stepCount
-    : !!challenge?.is_complete
+  const isComplete = !!challenge?.is_complete
```

Comment:
not that we need to consolidate this now, but this is ripe for some sort of ChallengeState type: `incomplete | inProgress | completeUndisbursed | disbursed` instead of all these booleans

Reply:
🤯